File: conexiones/conexion.go/postgres.go
// instalar go get -u github.com/go-sql-driver/mysql //

package main

import (
	"database/sql"
	"fmt"
	"log"

	_ "github.com/lib/pq"
)

func main() {
	connStr := "user=postgres password=1234 host=localhost port=5432 dbname=postgres sslmode=disable"

	db, err := sql.Open("postgres", connStr)
	if err != nil {
		log.Fatal("Error al abrir la conexión:", err)
	}
	defer db.Close()

	err = db.Ping()
	if err != nil {
		log.Fatal("Error al hacer ping a la base de datos:", err)
	}

	fmt.Println("Conexión exitosa")
}

File: conexiones/conexion.php/postgres.php
<?php
$host = "localhost";
$port = 5432;
$dbname = "postgres";
$user = "postgres";
$password = "1234";

// Crear la cadena de conexión
$conn_str = "host=$host port=$port dbname=$dbname user=$user password=$password";

// Intentar conectar a la base de datos
$conn = pg_connect($conn_str);

// Comprobar si la conexión fue exitosa
if ($conn) {
    echo "Conexión exitosa";
} else {
    echo "Error en la conexión";
}

// Mostrar la conexión (solo para depuración)
var_dump($conn);
?>
